lib/ocpp/v201/smart_charging.cpp

```diff
@@ -52,6 +53,14 @@ std::string profile_validation_result_to_string(ProfileValidationResultEnum e) {
         return "ChargingSchedulePeriodExtraneousPhaseValues";
     case ProfileValidationResultEnum::DuplicateTxDefaultProfileFound:
         return "DuplicateTxDefaultProfileFound";
+    case ProfileValidationResultEnum::InvalidProfileType:
```

Comment:
I wish we had a better way to get the string value. Boost provides BOOST_PP_STRINGIZE

If we could use C++17, Magic Enum looks like it could work?

A simple approach would be to declare a function using macros that returns a string value for an enum value, but it has downsides, like not handling variables as well as constant expressions.